SickChill: Update to v2024.3.1

mreid-tt opened the pull request:
## Description

This PR includes the following:

1. Update SickChill to v2024.3.1

Fixes # <!--Optionally, add links to existing issues or other PR's-->

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relevant tags.-->
- [x] Package update

hgy59 commented on spk/sickchill/Makefile:
```diff
@@ -52,7 +52,7 @@ WHEELS += src/requirements-crossenv-greenlet-v3-gcc4.txt
 else ifeq ($(call version_lt, $(TC_GCC), 4.9),1)
 # archs without C++11 support are not supported:
-UNSUPPORTED_ARCHS = $(ARCHS)
+UNSUPPORTED_ARCHS += $(ARCH)
```
seems obsolete, we already have

```
# archs without C++11 support are not supported [greenlet]
UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(OLD_PPC_ARCHS)
```

mreid-tt replied:
Can be removed when the package is migrated to python312

hgy59 commented:
> Just a comment. This has been edited from #6007 and merged within a few hours without any heads up or a chance to do any testing by the SickChill community.

+1 
Don't hurry, we must not reach short PR times!
It is important that another pair of eyes has a look at those. Even when it takes weeks...

mreid-tt commented:
Thanks for the feedback. I’ll definitely be more thoughtful going forward. My reasoning was that since testing was already done with #6007, it could be merged. The changes in this PR are fairly minor, and I’ve outlined them below:  

### Makefile Changes:
- Explicitly added `hi3535` to excluded architectures based on findings from `python311-wheels` for greenlet.  
- Switched cryptography to build directly instead of using `cross/cryptography`.  
- Combined the architecture check for greenlet into the same logic block, following the `python311-wheels` example.  
- Updated the changelog and corrected a spelling error in the comment before the ENV.  

### `src/requirements-crossenv.txt` Changes:
- Upgraded packages:  
  - `charset_normalizer`: 3.4.0 -> 3.4.1  
  - `lxml`: 5.3.0 -> 5.3.2  
  - `SQLAlchemy`: 2.0.36 -> 2.0.40  
  - `tornado`: 6.4.1 -> 6.4.2  
  - Added `cryptography` 44.0.2 (due to direct build).  

### `src/requirements-pure.txt` Changes:
- Upgraded packages:  
  - `beautifulsoup4`: 4.12.3 -> 4.13.3  
  - `cachecontrol`: 0.14.1 -> 0.14.2  
  - `click`: 8.1.7 -> 8.1.8  
  - `decorator`: 5.1.1 -> 5.2.1  
  - `dogpile.cache`: 1.3.3 -> 1.3.4  
  - `jsonrpclib-pelix`: 0.4.3.3 -> 0.4.3.4  
  - `Mako`: 1.3.6 -> 1.3.9  
  - `markdown2`: 2.5.1 -> 2.5.3  
  - `pbr`: 6.1.0 -> 6.1.1  
  - `putio.py`: 8.7.0 -> 8.8.0  
  - `pyOpenSSL`: 23.2.0 -> 24.3.0  
  - `pytz`: 2024.2 -> 2025.2  
  - `qbittorrent-api`: 2024.10.68 -> 2024.12.71  
  - `stevedore`: 5.3.0 -> 5.4.1  
  - `typing_extensions`: 4.12.2 -> 4.13.1  
  - `tzlocal`: 5.2 -> 5.3.1  
  - `urllib3`: 2.2.3 -> 2.3.0  

### Testing and Validation:
I followed the checklist for installation, reinstallation, start/stop, and conducted in-app testing, including:  
1. Configuring library and download folders.  
2. Setting up API connection to SABnzbd.  
3. Configuring the NZB indexing server connection.  
4. Adding existing series to the library and importing.  
5. Triggering a download for a missing episode.  
6. Verifying that the downloaded file was identified, moved, and renamed correctly in the library.  

All installation and service logs looked good, and the app showed no warnings or errors during my in-app testing.
